pennylane/resource/resource.py

```diff
@@ -55,21 +53,25 @@ class Resources:
     shots: int = 0
 
     def __str__(self):
-        keys = ["wires", "gates", "depth", "shots", "gate_types"]
-        vals = [self.num_wires, self.num_gates, self.depth, self.shots, self.gate_types]
+        keys = ["wires", "gates", "depth", "shots"]
+        vals = [self.num_wires, self.num_gates, self.depth, self.shots]
         items = "\n".join([str(i) for i in zip(keys, vals)])
         items = items.replace("('", "")
         items = items.replace("',", ":")
         items = items.replace(")", "")
-        items = items.replace("{", "\n{")
+
+        gate_str = ", ".join(
+            [f"'{gate_name}': {count}" for gate_name, count in self.gate_types.items()]
+        )
+        items += "\ngate_types:\n{" + gate_str + "}"
         return items
 
     def _ipython_display_(self):
         """Displays __str__ in ipython instead of __repr__"""
         print(str(self))
 
 
-def _count_resources(tape: QuantumTape, shots: int) -> Resources:
```

Comment:
type hint was removed due to circular import errors. We probably don't need it since its a private function

Reply:
You can also use inverted commas for type hinting :+1:

Reply:
Circular import errors are a side effect of circular dependencies. So OK in the short term, but we should consider an architecture going forward that gets rid of any circular dependencies. Tight coupling makes fragile code.